src/Dfc.CourseDirectory.WebV2/SharedViews/Components/AdminTopNav.cshtml

```diff
@@ -1,24 +1,23 @@
 ﻿@inject IFeatureFlagProvider FeatureFlagProvider
 @inject Settings Settings 
 @{
-    var currentController = (string)ViewContext.RouteData.Values["Controller"];
-    var currentAction = (string)ViewContext.RouteData.Values["Action"];
+    var activeSection = ViewData["ActiveAdminTopNavSection"] as AdminTopNavSection?;
 }
 
 <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
 <nav>
     <ul id="navigation" class="govuk-header__navigation" aria-label="Top Level Navigation">
-        <li class="govuk-header__navigation-item@(currentController == "HelpdeskDashboard" && currentAction == "Dashboard" ? " govuk-header__navigation-item--active" : null)">
+        <li class="govuk-header__navigation-item@(activeSection == AdminTopNavSection.Home ? " govuk-header__navigation-item--active" : null)">
```

Comment:
I'm sure I've seen this done before with a `TagHelper` of some sort...?

Reply:
You mean the active section detection?

Reply:
Possibly. It's been a while and I've not looked into it. Let's discuss tomorrow.